File: apps/mobile-portfolio/src/types/portfolio.ts
export type CurrencyCode = "USD" | "EUR" | "CNY" | "JPY" | "HKD";

export type AccountType =
  | "brokerage"
  | "saving"
  | "checking"
  | "creditCard"
  | "cash";

export interface Holding {
  id: string;
  symbol: string;
  name: string;
  quantity: number;
  price: number;
  costBasis: number;
  marketValue: number;
  profit: number;
  profitRate: number;
  assetClass: "equity" | "bond" | "cash" | "fund" | "other";
  riskLevel: "low" | "medium" | "high";
}

export interface Account {
  id: string;
  name: string;
  type: AccountType;
  currency: CurrencyCode;
  balance: number;
  todayChange: number;
  holdings: Holding[];
}

export interface PortfolioSummary {
  totalAssets: number;
  totalLiabilities: number;
  netWorth: number;
  todayChange: number;
  weekChange: number;
}

export interface PortfolioHistoryPoint {
  date: string;
  value: number;
}

export interface Portfolio {
  id: string;
  ownerName: string;
  baseCurrency: CurrencyCode;
  accounts: Account[];
  summary: PortfolioSummary;
  history: PortfolioHistoryPoint[];
}

File: apps/server-python/src/core/domain/entities/portfolio.py
from __future__ import annotations

from dataclasses import dataclass
from typing import List

from src.core.domain.value_objects.portfolio import HistoryPoint, PortfolioSummary


@dataclass
class Holding:
  id: str
  symbol: str
  name: str
  quantity: float
  price: float
  cost_basis: float
  market_value: float
  profit: float
  profit_rate: float
  asset_class: str
  risk_level: str


@dataclass
class Account:
  id: str
  name: str
  type: str
  currency: str
  balance: float
  today_change: float
  holdings: List[Holding]


@dataclass
class Portfolio:
  id: str
  owner_name: str
  base_currency: str
  accounts: List[Account]
  summary: PortfolioSummary
  history: List[HistoryPoint]
